src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -100,8 +114,60 @@ class AmendmentTableImpl : public AmendmentTable
 void
 AmendmentTableImpl::addInitial ()
 {
-    // For each amendment this version supports, construct the AmendmentState object by calling
-    // addKnown. Set any vetoes or defaults. A pointer to the AmendmentState can be stashed
+    // For each amendment this version supports, construct the AmendmentState
+    // object by calling addKnown. Set any vetoes or defaults. A pointer to the
+    // AmendmentState can be stashed
+
+    std::set<std::pair<uint256, std::string>> toAdd;
+
+    for (auto const& _ : detail::PreEnabledAmendments)
+    {
+        toAdd.emplace (uint256 (_.first), _.second);
+    }
+
+    {
+        // add the amendments from the config file
+        auto const& section = getConfig ().section (SECTION_AMENDMENTS);
+        int const numExpectedToks = 2;
+        for (auto const& _ : section.lines ())
+        {
+            boost::tokenizer<> tokenizer (_);
+            int numToks = 0;
+            uint256 id;
+            std::string friendlyName;
+            for(auto const& curTok : tokenizer)
+            {
+                ++numToks;
+                if (numToks > numExpectedToks)
+                {
+                    continue;  // continue counting so can report number in the
+                               // error
+                }
+
+                if (numToks == 1)
+                    id.SetHex (curTok);
+                else
+                    friendlyName = curTok;
+            }
+
```

Comment:
It may be worth asking how friendly the name needs to be.  Oftentimes names containing whitespace are more trouble than they are worth.  David or Vinnie may have an opinion on that.